### PR TITLE
View predictions using lidarseg devkit

### PR DESCRIPTION
**Pull request by whyekit-aptiv**
### This PR will:

- Allow users to view their lidarseg predictions using the devkit by adding the `show_lidarseg_preds` argument to the various functions related to lidarseg. Users will simply need to pass in the path to the folder which contains their predictions.

- Update the lidarseg tutorial to reflect the ability for users to view their predictions using the devkit.

### CODE REVIEW

File: python-sdk/nuscenes/nuscenes.py [hunk 1]
```diff
@@ -393,13 +393,17 @@ def box_velocity(self, sample_annotation_token: str, max_time_diff: float = 1.5)
         else:
             return pos_diff / time_diff
 
-    def get_sample_lidarseg_stats(self, sample_token: str, sort_counts: bool = True) -> None:
+    def get_sample_lidarseg_stats(self, sample_token: str, sort_counts: bool = True,
+                                  show_lidarseg_preds: str = None) -> None:
```
Comment (holger-nutonomy) [inline python-sdk/nuscenes/nuscenes.py:397]: show_lidarseg_preds - change the name to `lidarseg_pred_dir`

Comment (holger-nutonomy) [inline python-sdk/nuscenes/nuscenes.py:397]: Put each argument in one line

File: python-sdk/nuscenes/nuscenes.py [hunk 2]
```diff
@@ -737,7 +763,16 @@ def map_pointcloud_to_image(self,
         elif show_lidarseg_labels:
             assert pointsensor['sensor_modality'] == 'lidar', 'Error: Can only render lidarseg labels for lidar, ' \
                                                               'not %s!' % pointsensor['sensor_modality']
-            lidarseg_labels_filename = osp.join(self.nusc.dataroot, 'lidarseg', pointsensor_token + '_lidarseg.bin')
+
+            if show_lidarseg_preds:
+                sample_token = self.nusc.get('sample_data', pointsensor_token)['sample_token']
+                lidarseg_labels_filename = osp.join(show_lidarseg_preds, pointsensor_token + '.bin')
```
Comment (holger-nutonomy) [inline python-sdk/nuscenes/nuscenes.py:769]: Standardize file format to be the same as for GT

File: python-sdk/tutorials/nuscenes_lidarseg_tutorial.ipynb [hunk 3]
```diff
@@ -21,19 +21,43 @@
     "- ```lidarseg_category.json```\n",
     "- ```lidarseg.json```\n",
     "\n",
-    "Copy the ```lidarseg``` folder (this contains .bin files; each .bin file is an annotation for a pointcloud) into ```/data/sets/nuscenes/```, and copy ```lidarseg_category.json``` and ```lidarseg.json``` into ```/data/sets/nuscenes/v1.0-mini```.\n",
+    "Copy the `lidarseg` folder (this contains `.bin` files; each `.bin` file is an annotation for a pointcloud) into `/data/sets/nuscenes/`, and copy `lidarseg_category.json` and `lidarseg.json` into `/data/sets/nuscenes/v1.0-mini`.\n",
```
Comment (holger-nutonomy) [inline python-sdk/tutorials/nuscenes_lidarseg_tutorial.ipynb:24]: Let's remove the trailing slash from `/data/sets/nuscenes/`.

Comment (holger-nutonomy) [reply python-sdk/tutorials/nuscenes_lidarseg_tutorial.ipynb:24]: Remove `and copy `lidarseg_category.json` and `lidarseg.json` into `/data/sets/nuscenes/v1.0-mini`` and provide the .zip file with those subfolders.

File: python-sdk/tutorials/nuscenes_lidarseg_tutorial.ipynb [hunk 4]
```diff
@@ -21,19 +21,43 @@
...
+    "    │\n",
+    "    ├── v1.0-test\n",
+    "    │   ├── Usual files (e.g. attribute.json, calibrated_sensor.json etc.)     \n",
+    "    │   ├── lidarseg.json  <- contains the mapping of each .bin file to the token  \n",
+    "    │   └── lidarseg_category.json <- contains the categories of the labels\n",
+    "    │\n",
+    "    └── v1.0-trainval\n",
+    "        ├── Usual files (e.g. attribute.json, calibrated_sensor.json etc.)     \n",
+    "        ├── lidarseg.json  <- contains the mapping of each .bin file to the token  \n",
+    "        └── lidarseg_category.json <- contains the categories of the labels\n",
```
Comment (holger-nutonomy) [inline python-sdk/tutorials/nuscenes_lidarseg_tutorial.ipynb:60]: Let's remove all but one and make it  `v1.0-{mini, trainval, test}`

File: python-sdk/tutorials/nuscenes_lidarseg_tutorial.ipynb [hunk 5]
```diff
@@ -363,6 +387,44 @@
...
+    "                                show_lidarseg_legend=True,\n",
+    "                                verbose=True,\n",
+    "                                show_lidarseg_preds='/path/to/your/lidarseg_preds')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Note that each set of predictions in the folder must be a `.bin` file and named as `<lidar_sample_data_token>.bin`."
```
Comment (holger-nutonomy) [inline python-sdk/tutorials/nuscenes_lidarseg_tutorial.ipynb:425]: Change file name convention

File: python-sdk/tutorials/nuscenes_lidarseg_tutorial.ipynb [hunk 6]
```diff
@@ -363,6 +387,44 @@
...
    ]
   },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Visualizing LIDAR segmentation predictions\n",
+    "In all the above functions, the labels of the LIDAR pointcloud which have been rendered are the ground truth. If you have trained a model to segment LIDAR pointclouds and have run it on the nuScenes-lidarseg dataset, you can visualize your model's predictions simply by using the `show_lidarseg_preds` argument in the above functions.\n",
+    "\n",
+    "You simply need to pass the path to the folder where your predictions are stored into `show_lidarseg_preds`, like so:"
```
Comment (holger-nutonomy) [inline python-sdk/tutorials/nuscenes_lidarseg_tutorial.ipynb:397]: Explain that here we will do it with the ground-truth, but that the files are identical.

File: python-sdk/nuscenes/nuscenes.py (hunk 2 above)
Comment (whyekit-aptiv) [reply python-sdk/nuscenes/nuscenes.py]: This is no longer relevant as users now input the entire file path to the .bin file which contains their preds

**Pull request merged**